spynnaker/pyNN/models/recording_common.py

```diff
@@ -115,42 +172,46 @@ def _get_recorded_variable(self, variable):
         sim = get_simulator()
 
         globals_variables.get_simulator().verify_not_running()
-        if variable == "spikes":
-            data = self._get_spikes()
 
         # check that we're in a state to get voltages
-        elif not isinstance(
+        if not isinstance(
                 self._population._vertex, AbstractNeuronRecordable):
             raise ConfigurationException(
                 "This population has not got the capability to record {}"
                 .format(variable))
-        elif not self._population._vertex.is_recording(variable):
+
+        if not self._population._vertex.is_recording(variable):
             raise ConfigurationException(
                 "This population has not been set to record {}"
                 .format(variable))
 
-        elif not sim.has_ran:
+        if not sim.has_ran:
             logger.warn(
                 "The simulation has not yet run, therefore {} cannot"
                 " be retrieved, hence the list will be empty".format(
                     variable))
             data = numpy.zeros((0, 3))
```

Comment:
No worse than it used to be - however gsyn is (0, 4) in PyNN 7...

Reply:
Yes and NO.
You are correct that in Pynn0.7 gsyn is (0,4) BUT
In sPyNNaker7/spynnaker7/pyNN/models/pynn_population.py method get_gsyn this method was and still is called twice once for gsyn_exc and once for gsyn_inh and then the results are merged.

Reply:
Ahh OK